autoprecompiles/src/lib.rs

```diff
@@ -457,13 +461,24 @@ pub fn build<A: Adapter>(
 
     let machine = convert_machine_field_type(machine, &A::into_field);
 
-    let apc = Apc::new(block, machine, column_allocator);
+    let apc = Apc::new(block.clone(), machine, column_allocator);
 
     if let Some(path) = apc_candidates_dir_path {
+        std::fs::create_dir_all(path).expect("Failed to create directory for APC candidates");
+
+        let ser_path_unopt = path
```

Comment:
Can you extract this into a function to avoid code-duplication and also avoid cloning the column allocator?

Reply:
Agree with extracting into function, but `Apc::new` consumes the column allocator, so this wouldn't help with cloning